lighthouse-cli/test/fixtures/visibility.html

```diff
@@ -0,0 +1,60 @@
+<!--
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+-->
+
+<html>
+  <head>
+    <meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1">
+    <title>Visibility</title>
+  </head>
+  <body>
+    now you see me
+    <script>
+      window.addEventListener('visibilitychange', e => console.log(document.visibilityState));
+
+      function waitForVisibilityChange() {
+        return new Promise(resolve => window.addEventListener('visibilitychange', function() {
+          window.removeEventListener('visibilitychange', this);
+          resolve();
+        }));
+      }
+
+      function waitForFirstPaint() {
+        return new Promise((resolve) => {
+          const observer = new PerformanceObserver((list) => {
+            if (!list.getEntries().length) return;
+            observer.disconnect();
+            resolve();
+          });
+          observer.observe({entryTypes: ['paint']});
+        });
+      }
+
+      async function toggle(openNumTimes) {
+        for (let i = 0; i < openNumTimes; i++) {
+          const coverWindow = window.open();
```

Comment:
ah yah this'll make a new tab.

occlusion would be tested with floating popup (and its dimensions would have to exceed those of the underlying window)

Reply:
my first pass was occlusion. It wasn't consistent across OS, and required offsets of the main window (`--window-position=5,5`) so it could be completely covered.